public/js/index.js

```diff
@@ -21,7 +21,12 @@ postForm.on('submit', function (e) {
 
   var reader = new FileReader();
   reader.onload = function (evt) {
-    var vrp = JSON.parse(evt.target.result);
+    var vrp;
+    try {
+      vrp = JSON.parse(evt.target.result);
+    } catch(e) {
+      alert("Le fichier fourni n'est pas au format JSON");
```

Comment:
Do we call this function for the file submitted by the user or for the solution returned by the API? 

If it is for the files submitted by the user, is there any useful information in `e`, does it tell why parsing failed -- like, extra comma in line 99 for example? If that is the case, it might help the user to debug their JSON file.

I also suggest adding "valide" to the message "Le fichier fourni n'est pas au format JSON valide"  but this might be just my personal pref..

Reply:
We call this function for file submitted by the user.

For my case, error returned was `JSON.parse: unexpected character at line 8002 column 35 of the JSON data`, while problem was one field was blank. I am not sure JSON parse error are very explicit. 
I run some other tests and you are right, in some cases it is explicit. I will integrate JSON parse error to a clear error message.

I agree with 'valide'. I wrote it in my English version and then realize I needed to write in French :)

Reply:
The language of the front is another thing we need to fix at some point :/

Reply:
I aggree : https://gitlab.com/mapotempo/optimizer-api/-/issues/670

I think this is among most prioritary things. At least having an English version only instead of French version only?